### PR TITLE
fix: Cohere inconsistent embeddings and documents lengths

### PR DESCRIPTION
**Pull request by sjrl**
I am looking into this issue https://github.com/deepset-ai/haystack-core-integrations/issues/283 and found a bug that I think is unrelated. 

**Update:**
Running the code in issue https://github.com/deepset-ai/haystack-core-integrations/issues/283 shows that the changes in this branch do seem to help. See comment [here](https://github.com/deepset-ai/haystack-core-integrations/issues/283#issuecomment-1914775276).

Basically we were copying the embeddings so we'd end up with twice as many embeddings as documents. I don't think this will fix the error brought up in issue https://github.com/deepset-ai/haystack-core-integrations/issues/283, but we should avoid doing this. 

Additional:
* also fixed some failing integration tests

### CODE REVIEW

**Comment by sjrl** [comment]
Hey @ZanSara I'm noticing that the integration tests that require a Cohere API key are not running in CI. Is this on purpose or should we be able to run them?

**Comment by ZanSara** [comment]
Hey @sjrl, thank you for this PR! Let's split the two things though:
1. the bugfix in one PR, so I can approve it right away, and
2. issue + PR for the Cohere tests, we need to address that separately and other tests might need fixing
Thank you for your help!

**Review by ZanSara — APPROVED** [review]
Testing out the example code the improvement is evident! Thank you for fixing this one! :hugs:

**Comment by illorca-verbi** [comment]
Thanks for this fix. When will the release be cut for https://pypi.org/project/cohere-haystack/?

**Comment by anakin87** [comment]
released: https://pypi.org/project/cohere-haystack/0.3.1/
